test/unit/controllers/declaration/BorderTransportControllerSpec.scala

```diff
@@ -101,69 +100,70 @@ class BorderTransportControllerSpec extends ControllerSpec {
-      def declarationFactory() = aDeclaration(withType(DeclarationType.STANDARD))
-      behave like borderTransportController(declarationFactory)
-
-      "return 303 (SEE_OTHER) to TransportPayment" when {
         "valid options are selected" in {
           withNewCaching(declarationFactory())
 
           val correctForm = formData(IMOShipIDNumber, "SHIP001", "United Kingdom")
 
-          val result = controller.submitForm(Mode.Draft)(postRequest(correctForm))
+          val result = controller.submitForm(Mode.Normal)(postRequest(correctForm))
 
-          await(result) mustBe aRedirectToTheNextPage
-          thePageNavigatedTo mustBe controllers.declaration.routes.TransportPaymentController
-            .displayPage(Mode.Draft)
+          status(result) must be(SEE_OTHER)
+          redirectLocation(result) mustBe Some(controllers.routes.StartController.displayStartPage.url)
         }
-
       }
-    }
-    "we are on simplified declaration journey" should {
-      def declarationFactory() = aDeclaration(withType(DeclarationType.SIMPLIFIED))
-      behave like borderTransportController(declarationFactory)
 
-      "return 303 (SEE_OTHER) to TransportPayment" when {
-        "valid options are selected" in {
-          withNewCaching(declarationFactory())
+    }
 
-          val correctForm = formData(IMOShipIDNumber, "SHIP001", "United Kingdom")
+  "Transport Details Controller" when {
+    "we are on supplementary declaration journey" should {
+      def declarationFactory() = aDeclaration(withType(DeclarationType.SUPPLEMENTARY))
+      behave like borderTransportController(declarationFactory, controllers.declaration.routes.TransportContainerController.displayContainerSummary())
+    }
 
-          val result = controller.submitForm(Mode.Draft)(postRequest(correctForm))
+    "we are on standard declaration journey" should {
+      def declarationFactory() = aDeclaration(withType(DeclarationType.STANDARD))
+      behave like borderTransportController(declarationFactory, controllers.declaration.routes.TransportPaymentController.displayPage())
+    }
 
-          await(result) mustBe aRedirectToTheNextPage
-          thePageNavigatedTo mustBe controllers.declaration.routes.TransportPaymentController
-            .displayPage(Mode.Draft)
-        }
+    "we are on simplified declaration journey" should {
+      def declarationFactory() = aDeclaration(withType(DeclarationType.SIMPLIFIED))
+      behave like borderTransportControllerForInvalidJourney(declarationFactory)
+    }
 
-      }
+    "we are on occasional declaration journey" should {
+      def declarationFactory() = aDeclaration(withType(DeclarationType.OCCASIONAL))
+      behave like borderTransportControllerForInvalidJourney(declarationFactory)
     }
 
+    "we are on clearance declaration journey" should {
+      def declarationFactory() = aDeclaration(withType(DeclarationType.CLEARANCE))
+      behave like borderTransportControllerForInvalidJourney(declarationFactory)
+    }
```

Comment:
Should we use our helper `onJourney(...)`?

Reply:
I just went with fixing / extending whatever style was there already.  I could have a go at refactoring I guess as we are not too rushed.